modules.d/99memstrack/module-setup.sh

```diff
@@ -20,9 +20,15 @@ install() {
     inst "/bin/memstrack" "/bin/memstrack"
 
     inst "$moddir/memstrack-start.sh" "/bin/memstrack-start"
-    inst_hook cleanup 99 "$moddir/memstrack-report.sh"
-
     inst "$moddir/memstrack.service" "$systemdsystemunitdir/memstrack.service"
-
     $SYSTEMCTL -q --root "$initdir" add-wants initrd.target memstrack.service
+
+    pid=$$
+    if grep -wq "kdumpbase" <<< $(cat /proc/"$pid"/cmdline | tr '\0' ' '); then
+        inst "$moddir/memstrack-report.sh" "/bin/memstrack-report"
+        inst "$moddir/memstrack-report.service" "$systemdsystemunitdir/memstrack-report.service"
+        $SYSTEMCTL -q --root "$initdir" add-wants initrd.target memstrack-report.service
```

Comment:
This makes initrd.target want both memstrack-report.service and memstrack.service? Won't this make memstrack stop right after getting started?

Reply:
Yes indeed, memstrack will stop right after getting started, so the time interval for memstrack to live is very short. I do captured some data, however I'm not very certain if the longer time for memstrack to live, the better report it can give. From your comments, I agree this solution is no longer appropriate.